### PR TITLE
Refactor cloud provisioning

### PR DESCRIPTION
**Pull request by rvelaVenafi**
- Simplifies struct organization and hierarchy
- Adds constants for a lot of values
- Updates examples

### CODE REVIEW

File: cmd/vcert/cmdCloudKeystores.go [hunk 1]
```diff
@@ -76,19 +76,24 @@ func doCommandProvisionCloudKeystore(c *cli.Context) error {
...
+		result.GcpID = metadata.CertificateID
+		result.GcpName = metadata.CertificateName
+	default:
+		return fmt.Errorf("unknown keystore metadata type: %s", metadata.CloudKeystoreType)
 	}
 
-	result.MachineIdentityId = metadata.GetMachineIdentityMetadata().GetID()
-	result.MachineIdentityActionType = metadata.GetMachineIdentityMetadata().GetActionType()
-
-	err = result.Flush(flags.provisionFormat, flags.provisionOutputFile)
```
Comment (luispresuelVenafi) [inline cmd/vcert/cmdCloudKeystores.go:90]: Why did you remove this? we need this for --file flag to work

Comment (luispresuelVenafi) [inline cmd/vcert/cmdCloudKeystores.go:90]: Why did you remove this? we need this for --file flag to work

**Review by luispresuelVenafi — CHANGES_REQUESTED** [review]

File: cmd/vcert/cmdCloudKeystores.go (hunk 1 above)
Comment (rvelaVenafi) [reply cmd/vcert/cmdCloudKeystores.go:90]: This got lost during branch rebase. Added back.

**Review by luispresuelVenafi — APPROVED** [review]

**Review by marcos-albornoz — APPROVED** [review]
